src/Policies.cpp

```diff
@@ -256,6 +259,10 @@ std::vector<Policy *> PolicyManager::load(const YAML::Node &policy_yaml)
             throw PolicyException(fmt::format("policy [{}] creation failed (policy): {}", policy_name, e.what()));
         }
         try {
+            //Avoid deadlock in input manager
```

Comment:
@weyrick  check if you agree with latest changes. moved the policy add to lock context and release the mutex before calling input_manager again

Reply:
Ok this is the right idea, but the problem is we still have a chance of a race between the unlock() and the module_exists/module_add call. I think what we should do here for certainly is make the module_* functions which need a unique_lock take an optional argument where we can move this lock in, and it can use it. That will then guarantee that there will be no races.